integration/spark2/src/main/scala/org/apache/spark/sql/CarbonCatalystOperators.scala

```diff
@@ -86,6 +87,29 @@ case class ShowLoadsCommand(databaseNameOp: Option[String], table: String, limit
   }
 }
 
+case class ProjectForUpdate(
+    table: UnresolvedRelation,
+    columns: List[String],
+    child: Seq[LogicalPlan] ) extends Command {
```

Comment:
It cannot extend a command as it required to be analyzed in analysis phase so it should extend LogicalPlan in 2.1